clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -52,6 +59,118 @@ abstract class OpenstackClientProvider {
   }
 
   /**
+   * Associate an instance with a load balancer.
+   * @param lbPoolId
+   * @param instanceId
+   */
+  void registerInstancesWithLoadBalancer(String lbPoolId, String instanceId) {
+
+    //weight of member
+    int weight = 1
+
+    //get pool
+    LbPool pool = getPool(lbPoolId)
+
+    //port from load balancer pool
+    int internalPort = getInternalLoadBalancerPort(pool)
+
+    //lookup instance ip
+    String ip = getIpForInstance(instanceId)
+
+    //add to pool
+    //TODO use handleRequest once that is refactored
+    client.networking().loadbalancers().member().create(
+      Builders.member().address(ip).poolId(lbPoolId).protocolPort(internalPort).weight(weight).build()
+    )
+  }
+
+  /**
+   * Unassociate an instance with a load balancer.
+   * @param lbPoolId
+   * @param instanceId
+   */
+  void deregisterInstancesFromLoadBalancer(String lbPoolId, String instanceId) {
+
+    //lookup instance ip
+    String ip = getIpForInstance(instanceId)
+
+    //get pool
+    LbPool pool = getPool(lbPoolId)
+
+    //get member with matching IP
+    //TODO use handleRequest once that is refactored
+    String memberId = client.networking().loadbalancers().member().list().find { m -> m.address == ip }?.id
```

Comment:
That is what the member api allows. Even in the UI there is a 'member' tab that shows all member instances for all load balancers.